### PR TITLE
[FIX] Make `fetch_atlas_aal` work with `SPM5` and `SPM8`

### LINKED ISSUE #3087: `fetch_atlas_aal` download error with `SPM5` and `SPM8`
With Nilearn 0.8.2.dev.
I haven't investigated further than trying all possible versions with cleaned data cache, so I might have missed something.
It seems to work for `SPM12` (the default), but fails on my machine with all other versions.

Code to reproduce:

```python
from nilearn import datasets
b = datasets.fetch_atlas_aal(version='SPM5')  # 'SPM8' gives same error
```
```
OSError                                   Traceback (most recent call last)
<ipython-input-97-6fd2a3e7006b> in <module>
----> 1 b = datasets.fetch_atlas_aal(version='SPM5')

~/GitRepos/nilearn-fork/nilearn/datasets/atlas.py in fetch_atlas_aal(version, data_dir, url, resume, verbose)
    938     see :footcite:`AAL_atlas`,
    939     and :footcite:`TZOURIOMAZOYER2002273`.
--> 940 
    941     Parameters
    942     ----------

~/GitRepos/nilearn-fork/nilearn/datasets/utils.py in _fetch_files(data_dir, files, resume, verbose, session)
    706         with requests.Session() as session:
    707             session.mount("ftp:", _NaiveFTPAdapter())
--> 708             return _fetch_files(
    709                 data_dir, files, resume=resume,
    710                 verbose=verbose, session=session)

~/GitRepos/nilearn-fork/nilearn/datasets/utils.py in _fetch_files(data_dir, files, resume, verbose, session)
    784             if os.path.exists(temp_dir):
    785                 shutil.rmtree(temp_dir)
--> 786             raise IOError('Fetching aborted: ' + abort)
    787         files_.append(target_file)
    788     # If needed, move files from temps directory to final directory.

OSError: Fetching aborted: Not a gzipped file (b'\x1f\x9d')
> /home/nicolas/GitRepos/nilearn-fork/nilearn/datasets/utils.py(786)_fetch_files()
    784             if os.path.exists(temp_dir):
    785                 shutil.rmtree(temp_dir)
--> 786             raise IOError('Fetching aborted: ' + abort)
    787         files_.append(target_file)
    788     # If needed, move files from temps directory to final directory.
```

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Closes #3087 

To try it out, remove `aal` folders from `nilearn_data` and you should get the following output:

```python
In [1]: from nilearn import datasets, image
/home/nicolas/GitRepos/nilearn-fork/nilearn/datasets/__init__.py:93: FutureWarning: Fetchers from the nilearn.datasets module will be updated in version 0.9 to return python strings instead of bytes and Pandas dataframes instead of Numpy arrays.
  warn("Fetchers from the nilearn.datasets module will be "

In [2]: for version in [5, 8, 12]:
   ...:     bunch = datasets.fetch_atlas_aal(version=f"SPM{version}")
   ...:     assert len(bunch.description) > 0
   ...:     assert len(bunch.labels) == 116
   ...:     assert len(bunch.indices) == 116
   ...:     assert image.load_img(bunch.maps).shape ==  (91, 109, 91)
   ...: 

Dataset created in /home/nicolas/nilearn_data/aal_SPM5

Downloading data from http://www.gin.cnrs.fr/wp-content/uploads/aal_for_SPM5.zip ...
 ...done. (0 seconds, 0 min)
Extracting data from /home/nicolas/nilearn_data/aal_SPM5/f158e2c2ef1b2be86d7939837f482c25/aal_for_SPM5.zip..... done.

Dataset created in /home/nicolas/nilearn_data/aal_SPM8

Downloading data from http://www.gin.cnrs.fr/wp-content/uploads/aal_for_SPM8.zip ...
 ...done. (0 seconds, 0 min)
Extracting data from /home/nicolas/nilearn_data/aal_SPM8/8207ad831ac9f41247d3aeed52ddddb8/aal_for_SPM8.zip..... done.

Dataset created in /home/nicolas/nilearn_data/aal_SPM12

Downloading data from http://www.gin.cnrs.fr/AAL_files/aal_for_SPM12.tar.gz ...
 ...done. (2 seconds, 0 min)
Extracting data from /home/nicolas/nilearn_data/aal_SPM12/9845813e2300e5b614a9dd2713e13445/aal_for_SPM12.tar.gz..... done.
```

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, I confirm it works.

**Comment by NicolasGensollen** [comment]
@ymzayek do you think you could review this PR if you have the time?
This is the last one in the `0.9.1` milestone and I'd like to have another approval before merging.
Thanks!

**Comment by ymzayek** [comment]
@NicolasGensollen sure I will have a look in about an hour or so and will get back to you asap

**Comment by NicolasGensollen** [comment]
@ymzayek No worries, there is no rush ! ;)

**Review by ymzayek — APPROVED** [review]
I also confirm it's working, thx!

**Pull request merged**
